[TASK] Add components of TYPO3 extension (#4457)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ExtensionArchitecture/ComponentsOfTYPO3Extension/CreatingDatabaseModel.rst

> .. _creating-database-model:

Reviewer: ```suggestion
..  include:: /Includes.rst.txt
.. _creating-database-model:
```

> ===================================

Reviewer: ```suggestion
=============================
```

Creating a new database model

> ===================================

Reviewer: ```suggestion
=============================
```

> Create SQL Database Schema

Reviewer: ```suggestion
Create SQL database schema
```

---------------------------

* Add `example-extension/ext_tables.sql`

> * Create database schema

Reviewer: ```suggestion
* Insert your SQL database schema definition into that file
```

(Otherwise people may think that they need to create that on database-level and maybe the file would get automatically created from their database ;) )

  * Each entity is represented by one database table
    * Table name has the following structure: `tx_{extension key without underscores}_domain_model_{entity name}`
    * Each entity property is represented by one database column

.. code-block:: sql

    CREATE TABLE tx_exampleextension_domain_model_example (
        title       varchar(255)     DEFAULT ''  NOT NULL,
        description text             DEFAULT '',
        foo_foo     tinyint(1)       DEFAULT '1'
    );

> Create TCA Configuration

Reviewer: ```suggestion
Create TCA configuration
```

-------------------------

> * Add `./Configuration/TCA/{table name}.php`

Reviewer: ```suggestion
* Add `example-extension/Configuration/TCA/{table name}.php`
```

>   * In the example: `example-extension/Configuration/TCA/tx_exampleextension_domain_model_example.php`

Reviewer: Could we show the example of a TCA for the dummy properties set up in ext_tables.sql here?

> * The TCA configures types, validation and backend-UI-related stuff for each entity property

Reviewer: ```suggestion
* The TCA defines types, validation and backend-UI-related parameters for each entity property
```

>   * See `Documentation <https://docs.typo3.org/m/typo3/reference-tca/main/en-us/>`_ for further information

Reviewer: ```suggestion
  * See :ref:`TCA Reference <t3tca:start>` for further information
```

> * Add localization files

Reviewer: ```suggestion
* Add localization files according to :ref:`Language <t3coreapi:extension-Resources-Private-Language>`
```

>   * `./Resources/Private/Language/locallang_db.xlf`

Reviewer: ```suggestion
  * `example-extension/Resources/Private/Language/locallang_db.xlf`
```

>   * `./Resources/Private/Language/de.locallang_db.xlf`

Reviewer: ```suggestion
  * `example-extension/Resources/Private/Language/de.locallang_db.xlf`
```

* Add table icon

>   * `./Resources/Public/Icons/{entity name}.svg`

Reviewer: ```suggestion
  * `example-extension/Resources/Public/Icons/{entity name}.svg`
```

..  include:: /Includes.rst.txt
.. _creating-database-model:

=============================
Creating a new database model
=============================

Create SQL database schema
---------------------------

* Add `example-extension/ext_tables.sql`
* Insert your SQL database schema definition into that file
  * Each entity is represented by one database table
    * Table name has the following structure: `tx_{extension key without underscores}_domain_model_{entity name}`
    * Each entity property is represented by one database column

> .. code-block:: sql

Reviewer 2: For most field types we don't need to add them manually to the ext_tables.sql anymore in TYPO3 13. So I would say: Model the database scheme from the TCA or php model perspective, then check which fields still have to be added to the ext_tables regarding TYPO3 version?

Reviewer: That's a very good point, yes. There's really no great workflow for this. Maybe point to the "Auto-db create field" changelog? There we have a documentation which TCA types are auto-created. Or did we migrat that to a documentation page already, to indicate which types are autocreated?! Better to link there than to suggest a workflow like "check which fields are missing in your database, because you'll get errors for some fields but not for others", that might be very frustrating?

Author: Let me know if current version works for you.
